pandas/core/arrays/datetimelike.py

```diff
@@ -1299,8 +1299,16 @@ def __add__(self, other):
             # TimedeltaIndex, ndarray[timedelta64]
             result = self._add_delta(other)
         elif is_object_dtype(other):
-            # e.g. Array/Index of DateOffset objects
-            result = self._addsub_object_array(other, operator.add)
+            from pandas import Index
+
+            if type(other) is Index:
```

Comment:
i think we also want ndarray here.  so we want to exclude something like `isinstance(other, ABCIndex) and type(other) is not Index`